### ISSUE TITLE
steam running and showing desktop icon after exiting via icon/steam drop down menu

### ISSUE THREAD

**Issue by zooki2006** [ValveSoftware/steam-for-linux#8243]
#### Your system information

* Steam client version (build number or date): 
* Distribution (e.g. Ubuntu): arch linux(endeavor os)
* Opted into Steam client beta?: [Yes/No] No
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)
when i exit steam via the icon on taskbar or via the steam drop down menu it will close but seeming keep running in htop and the icon is still on my task bar but the right click options don't work i have to kill it via htop

also i ran it in the terminal and got this output
[zooki@zooki-arch-D ~]$ steam 
Running Steam on endeavouros rolling 64-bit
STEAM_RUNTIME is enabled automatically
Steam runtime environment up-to-date!
Steam client's requirements are satisfied
WARNING: Using default/fallback debugger launch
/home/zooki/.local/share/Steam/ubuntu12_32/steam
[2021-11-19 12:06:50] Startup - updater built Oct 13 2021 19:47:06
Installing breakpad exception handler for appid(steam)/version(1634158817)
[2021-11-19 12:06:50] Loading cached metrics from disk (/home/zooki/.local/share/Steam/package/steam_client_metrics.bin)
[2021-11-19 12:06:50] Using the following download hosts for Public, Realm steamglobal
[2021-11-19 12:06:50] 1. https://cdn.akamai.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-11-19 12:06:50] 2. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-11-19 12:06:50] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
Installing breakpad exception handler for appid(steam)/version(1634158817)
[2021-11-19 12:06:50] Verifying installation...
[2021-11-19 12:06:50] Verification complete
Loaded SDL version 2.0.17-6744061
Installing breakpad exception handler for appid(steam)/version(1634158817)
Runtime for steamwebhelper: defaulting to /home/zooki/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
CAppInfoCacheReadFromDiskThread took 63 milliseconds to initialize
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Proceed to auto login
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
src/clientdll/compatmanager.cpp (905) : GetCompatibilityToolCommandLineInternal on dependent tool failed
src/clientdll/compatmanager.cpp (905) : GetCompatibilityToolCommandLineInternal on dependent tool failed
Installing breakpad exception handler for appid(steam)/version(1634158817)
DBG DBG ffffffff DUMP_REQUESTED Assert( GetCompatibilityToolCommandLineInternal on dependent tool failed ):/data/src/clientdll/compatmanager.cpp:905

assert_20211119120651_22.dmp[36621]: Uploading dump (out-of-process)
/tmp/dumps/assert_20211119120651_22.dmp
src/clientdll/compatmanager.cpp (905) : GetCompatibilityToolCommandLineInternal on dependent tool failed
src/clientdll/compatmanager.cpp (905) : GetCompatibilityToolCommandLineInternal on dependent tool failed
Opted-in Controller Mask for AppId 0: f

(steam:36533): Gtk-WARNING **: 12:06:52.017: gtk_disable_setlocale() must be called before gtk_init()
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
assert_20211119120651_22.dmp[36621]: Finished uploading minidump (out-of-process): success = yes
assert_20211119120651_22.dmp[36621]: response: CrashID=bp-20db5e95-d4bf-4a8c-b981-b24052211119
assert_20211119120651_22.dmp[36621]: file ''/tmp/dumps/assert_20211119120651_22.dmp'', upload yes: ''CrashID=bp-20db5e95-d4bf-4a8c-b981-b24052211119''
roaming config store loaded successfully - 50420 bytes.
migrating temporary roaming config store
Installing breakpad exception handler for appid(steam)/version(1634158817)
BRefreshApplicationsInLibrary 1: 18ms
Installing breakpad exception handler for appid(steam)/version(1634158817)
Failed to init SteamVR because it isn't installed
ExecCommandLine: "'/home/zooki/.local/share/Steam/ubuntu12_32/steam'"
System startup time: 3.10 seconds
BuildCompleteAppOverviewChange: 686
RegisterForAppOverview 1: 18ms
RegisterForAppOverview 2: 19ms
Installing breakpad exception handler for appid(steam)/version(1634158817)
src/common/pipes.cpp (885) : stalled cross-thread pipe.
src/common/pipes.cpp (885) : stalled cross-thread pipe.
Installing breakpad exception handler for appid(steam)/version(1634158817)
DBG DBG ffffffff DUMP_REQUESTED Assert( stalled cross-thread pipe. ):/data/src/common/pipes.cpp:885

assert_20211119120707_31.dmp[36780]: Uploading dump (out-of-process)
/tmp/dumps/assert_20211119120707_31.dmp
src/clientdll/steamclient.cpp (854) : bufRet.TellPut() == sizeof(uint8)
src/clientdll/steamclient.cpp (854) : bufRet.TellPut() == sizeof(uint8)
assert_20211119120707_31.dmp[36780]: Finished uploading minidump (out-of-process): success = yes
assert_20211119120707_31.dmp[36780]: response: CrashID=bp-1a2f79f2-8410-4477-84ce-2baea2211119
assert_20211119120707_31.dmp[36780]: file ''/tmp/dumps/assert_20211119120707_31.dmp'', upload yes: ''CrashID=bp-1a2f79f2-8410-4477-84ce-2baea2211119''
Thread "CJobMgr::m_WorkThreadPool:0" (ID 36611) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:0" (ID 36611) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:0" (ID 36611) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:0" (ID 36611) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:0" (ID 36611) failed to shut down
Thread "CJobMgr::m_WorkThreadPool:0" (ID 36611) failed to shut down
^CExiting workitem thread

(steam:36533): GLib-GObject-CRITICAL **: 12:08:14.561: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(steam:36533): GLib-GObject-WARNING **: 12:08:14.563: invalid (NULL) pointer instance

(steam:36533): GLib-GObject-CRITICAL **: 12:08:14.563: g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(steam:36533): GLib-GObject-CRITICAL **: 12:08:14.563: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
[2021-11-19 12:08:14] Shutdown
[zooki@zooki-arch-D ~]$ 

TLDR
after full exiting the program via steam's drop down menu or taskbar icon it will keep running and in the termail it was run(if it was run from one) keep showing the error "Thread "CJobMgr::m_WorkThreadPool:0" (ID 36611) failed to shut down"

#### Steps for reproducing this issue:

1. run steam
2. exit steam via drop down menu or taskbar icon
3. run top/htop and see that steam is still runing and you have to force close it

**Comment by kisak-valve** [comment]
Hello @zooki2006, this issue is already being tracked at #8113 / #6811. Closing as a duplicate.
